src/components/transactions/TxDetails/TxData/OnChainConfirmation/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { Safe__factory } from '@/types/contracts'
+import { Link as MuiLink, Skeleton, Stack, Typography } from '@mui/material'
+import { type TransactionData } from '@safe-global/safe-gateway-typescript-sdk'
+import TxData from '..'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { MethodDetails } from '../DecodedData/MethodDetails'
+
+import MethodCall from '../DecodedData/MethodCall'
+import { Divider } from '@/components/tx/DecodedTx'
+import Link from 'next/link'
+import { useCurrentChain } from '@/hooks/useChains'
+import { AppRoutes } from '@/config/routes'
+import { useGetTransactionDetailsQuery } from '@/store/api/gateway'
+import { useMemo } from 'react'
+import { skipToken } from '@reduxjs/toolkit/query'
+
+export const OnChainConfirmation = ({ data }: { data?: TransactionData }) => {
+  const chain = useCurrentChain()
+  const signedHash = useMemo(() => {
+    const safeInterface = Safe__factory.createInterface()
+    const params = data?.hexData ? safeInterface.decodeFunctionData('approveHash', data?.hexData) : undefined
+    if (!params || params.length !== 1) {
+      return
+    }
+
+    const signedHash = params[0] as string
+    return signedHash
```

Comment:
To save the casting, we can add this to the early return.

```suggestion
    if (!params || params.length !== 1 || typeof params[0] !== 'string') {
      return
    }

    return params[0]
```